install_ffmpeg.sh

```diff
@@ -75,47 +75,88 @@ if [ ! -e "$ROOT/x264" ]; then
+          && curl -LO https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-${LIBTENSORFLOW_VERSION}.tar.gz \
+          && sudo tar -C $ROOT/compiled -xzf libtensorflow-cpu-linux-x86_64-${LIBTENSORFLOW_VERSION}.tar.gz \
+          && sudo ldconfig \
+          && rm libtensorflow-cpu-linux-x86_64-${LIBTENSORFLOW_VERSION}.tar.gz
+        fi
         echo "experimental tag detected, building with Tensorflow support"
         EXTRA_FFMPEG_FLAGS="$EXTRA_FFMPEG_FLAGS --enable-libtensorflow"
     fi
   fi
 fi
 
+if [[ $BUILD_TAGS == *"debug-video"* ]]; then
+    echo "video debug mode, building ffmpeg with tools, debug info and additional capabilities for running tests"
+    DEV_FFMPEG_FLAGS="--enable-demuxer=hls --enable-filter=ssim --enable-encoder=wrapped_avframe,pcm_s16le --enable-shared --enable-debug=3 --disable-stripping --disable-optimizations --enable-encoder=libx265,libvpx_vp8,libvpx_vp9 --enable-decoder=hevc,libvpx_vp8,libvpx_vp9 --enable-libx265 --enable-libvpx"
+    FFMPEG_MAKE_EXTRA_ARGS="-j4"
+fi
+
 if [ ! -e "$ROOT/ffmpeg/libavcodec/libavcodec.a" ]; then
   git clone https://github.com/livepeer/FFmpeg.git "$ROOT/ffmpeg" || echo "FFmpeg dir already exists"
   cd "$ROOT/ffmpeg"
   git checkout 682c4189d8364867bcc49f9749e04b27dc37cded
   ./configure ${TARGET_OS:-} --fatal-warnings \
-    --disable-programs --disable-doc --disable-sdl2 --disable-iconv \
+    --disable-doc --disable-sdl2 --disable-iconv \
     --disable-muxers --disable-demuxers --disable-parsers --disable-protocols \
     --disable-encoders --disable-decoders --disable-filters --disable-bsfs \
     --disable-postproc --disable-lzma \
     --enable-libx264 --enable-gpl \
     --enable-protocol=rtmp,file,pipe \
-    --enable-muxer=mpegts,hls,segment,mp4,null --enable-demuxer=flv,mpegts,mp4,mov \
-    --enable-bsf=h264_mp4toannexb,aac_adtstoasc,h264_metadata,h264_redundant_pps,extract_extradata \
-    --enable-parser=aac,aac_latm,h264 \
+    --enable-muxer=mpegts,hls,segment,mp4,hevc,matroska,webm,null --enable-demuxer=flv,mpegts,mp4,mov,webm,matroska \
+    --enable-bsf=h264_mp4toannexb,aac_adtstoasc,h264_metadata,h264_redundant_pps,hevc_mp4toannexb,extract_extradata \
+    --enable-parser=aac,aac_latm,h264,hevc,vp8,vp9 \
     --enable-filter=abuffer,buffer,abuffersink,buffersink,afifo,fifo,aformat,format \
     --enable-filter=aresample,asetnsamples,fps,scale,hwdownload,select,livepeer_dnn,signature \
-    --enable-encoder=aac,libx264 \
-    --enable-decoder=aac,h264 \
+    --enable-encoder=aac,opus,libx264 \
+    --enable-decoder=aac,opus,h264 \
     --extra-cflags="-I${ROOT}/compiled/include" \
     --extra-ldflags="-L${ROOT}/compiled/lib ${EXTRA_LDFLAGS}" \
     --prefix="$ROOT/compiled" \
-    $EXTRA_FFMPEG_FLAGS
-  make
+    $EXTRA_FFMPEG_FLAGS \
+    $DEV_FFMPEG_FLAGS
+fi
+
+if [[ ! -e "$ROOT/ffmpeg/libavcodec/libavcodec.a" || $BUILD_TAGS == *"debug-video"* ]]; then
+  cd "$ROOT/ffmpeg"
+  make $FFMPEG_MAKE_EXTRA_ARGS
```

Comment:
Shouldn't we run make unconditionally after running the ffmpeg configure script?

Reply:
Make will run with same condition as configure (static lib is missing), plus it will run always in debug mode for convenience, in case source files were edited.

Reply:
Ah! Right because that static lib wouldn't exist after the configure step and would only exist after make is run. Got it!